cmd/bucket-replication.go

```diff
@@ -477,6 +477,7 @@ func replicateObject(ctx context.Context, objInfo ObjectInfo, objectAPI ObjectLa
 			replicationStatus = replication.Failed
 		}
 		r.Close()
+		gr.Close()
```

Comment:
We use gr.Close() in many places instead of defer gr.Close() to lessen the lock acquiring time ?

Reply:
@vadmeste I didn't want to risk that something else was relying on the lock to be released earlier.

Since this isn't something I wrote, I am hesitant to just add it as defer.

Reply:
r.Close() is better as it it wraps around gr. MonitoredReader has a internal state to see if the reader is closed, would likely cause incorrect bandwidth measurement if gr.Close() used here.

Reply:
r.Close() is sufficient here since gr.Close is called by r.Close() - also we need to close here because there is a potential to acquire lock once more due to DeleteObject() subsequently below @vadmeste

Reply:
This magic is horrible: https://github.com/minio/minio/blob/8a16a1a1a9e69d8fdddc227d7701b3d534688c5d/pkg/bucket/bandwidth/reader.go#L80

Either accept a `io.ReadCloser` in params and take over closing OR don't do magic like this.

@poornas @harshavardhana

Reply:
yes we should do `m.reader.(io.Closer)` not `io.ReadCloser`